### PR TITLE
Opendrive automated vandv

### PR DESCRIPTION
**Pull request by bradleynull**
## Overview

This automates many canary tests that help debug the simulator and client. The tests in the `diagnostics` directory can be run with ctest to quickly diagnose any errors or bottles necks in local and distributed systems.

Additionally, the response message from the sending server commands has been updated to include the entire `ApiMessage` data structure for a more clean means of parsing responses from the server.

## Changes 

* Implemented several server command/response tests to ensure connectivity with the simulator and supported functionality
* Implemented several sensor tests to validate sensor outputs and performance
* Modified `sendCommand` to respond with an `ApiMessage`
* Fixed some of the lane following examples to use new `sendCommand`

## Testing
* run all diagnostic test with `ctest --verbose` from your `./build` directory
* test that the client fails gracefully when the server is unlicensed
* general evaluate logging approach (open to other suggestions or patterns)
* make sure that examples fail loudly then the file path or working directory is incorrect
* lane_follower/fixed_step
* lane_follower/fixed_step_lidar
* lane_follower/real_time
* lane_follower/speed_curve
* multi_process_example/vehicle_controller.cpp

### CODE REVIEW

**Comment by stephendb** [comment]
Change this to development, shadow_fix was merged last week. Otherwise it looks good. Also, the point projection method is pretty eh now with opendrive. Better to just take vehicles current position lane spline, then do a GetNext at some fixed distance and use that as the steering point. This will be a bit weird at junctions if you haven't determined a path but then again that is beyond the scope of this PR.

File: examples/ros/src/simulator_control/config/simulator.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
   "id": "simulator",
   "ros": {
     "port": 9090,
-    "server": "127.0.0.1"
+    "server": "192.168.2.11"
   },
-  "server_ip": "127.0.0.1",
+  "server_ip": "192.168.2.11",
```
Comment (stephendb) [inline examples/ros/src/simulator_control/config/simulator.json:7]: Need to swap these back

**Review by stephendb — CHANGES_REQUESTED** [review]
think just need to swap the server_ip back

File: examples/ros/src/simulator_control/config/simulator.json (hunk 1 above)
Comment (bradleynull) [reply examples/ros/src/simulator_control/config/simulator.json:7]: Nice catch! Fixed.

**Comment by bradleynull** [comment]
@stephendb Ready for another look.

**Review by stephendb — APPROVED** [review]

**Pull request merged**